provider/docker.go

```diff
@@ -656,3 +673,49 @@ func parseService(service swarmtypes.Service, networkMap map[string]*dockertypes
 	}
 	return dockerData
 }
+
+func listTasks(ctx context.Context, dockerClient client.APIClient, serviceID string,
+	serviceDockerData dockerData, networkMap map[string]*dockertypes.NetworkResource) ([]dockerData, error) {
+	taskList, err := dockerClient.TaskList(ctx, dockertypes.TaskListOptions{})
```

Comment:
You should be able to use filters here (and thus not get all tasks and having to iterate over it).

```go
filters := filters.NewArgs()
filters.Add("service", serviceID)
taskList, err := dockerClient.TaskList(ctx, dockertypes.TaskListOptions{Filters: filters})
// […]
```